### DISCUSSION TITLE
Auto Step Climb functionality for the A380X

### DISCUSSION THREAD

**Discussion by AFitzy17** [flybywiresim/aircraft#9245] [Ideas]
Hello, I hope this is the right place to suggest this.

First of all , big congratulations on the release of your A380X! Massive undertaking the whole team has had in delivering us such an excellent freeware product! Looking forward to where you go from here.

Really looking forward to doing long hauls in this thing. One thing I'd like to see which I've gotten used to from PMDG's 777 is using the auto step climb feature.

Basically just a toggle that will climb to the next step climb when you reach it. Now I'm not sure how it works for the A380 compared to the 777, as the 777 recommends when to climb to the next altitude. Does the A380 have similar functionality that you could tap into?

The only real reason I use auto step climb is for when I do a long haul whilst at work or sleeping overnight, so that the plane does it itself.

Just thought it'd be a cool thing for you guys to add down the line, this definitely isn't a high priority or anything but I saw you have similar features like the simrate toggle and pause at TOD so it'd be neat to have this too!

Thanks again for all your hard work

**Comment by infinisigma** [comment]
The dev team on Discord [mentioned](https://discord.com/channels/738864299392630914/1302583655977320468) that this is something they are considering, but "this goes against our philosophy of only implementing realistic features" since auto step climbs aren't in the real aircraft and flight crews have to manually change the altitude in the FCU to initiate the climb.

However, this would be a great QoL feature, and one that has been available in PMDG birds for a while, so while yes, it's not realistic, we should have the option, in my humble opinion.

**Reply by Fowl38** [reply]
I prefer to have the automatic step climb feature than the unrealistic IRS instant alignment feature.

**Reply by tracernz** [reply]
I'm not sure how that is relevant, as that's already implemented, thus no time required for it, and it's also necessary to have it for spawn states other than cold & dark on the apron anyway.

I'd also like automatic step climb and sim rate, but my time is finite and I choose to spend it on implementing other things for the foreseeable future. Anyone who feels strongly about this one over others can choose to implement it and submit a PR.
